README.md

```diff
@@ -941,6 +952,326 @@ In order to use your receiver you need a receiverId. This is the ID of your rece
+        return;
+      }
+
+      const frameDoc = playerIFrame.contentDocument;
+      if (!frameDoc) {
+        return;
+      }
+
+
+      const adsContainer = frameDoc.querySelector('.video-ads');
+      if (!adsContainer || adsContainer.childElementCount == 0) {
+        return;
+      }
+
+      const adsVideo = adsContainer.querySelector("video");
+
+      if (adsVideo) {
+        adsVideo.muted = true;
+        adsVideo.style.display = 'none';
+        adsVideo.currentTime = adsVideo.duration - 0.15;
+        adsVideo.muted = false;
+        adsVideo.style.display = '';
+        if (adblockIntervalId) {
+          clearInterval(adblockIntervalId);
+        }
+      }
+      else {
+        const isAdShowing = frameDoc.getElementsByClassName('ad-showing').length != 0;
+        if (!isAdShowing) {
+          return;
+        }
+
+        const mainVideo = frameDoc.querySelector('.html5-main-video');
+        if (!mainVideo) {
+          return;
+        }
+
+        mainVideo.muted = true;
+        mainVideo.currentTime = mainVideo.duration - 0.15;
+        mainVideo.muted = false;
+        if (adblockIntervalId) {
+          clearInterval(adblockIntervalId);
+        }
+      }
+    }, 100);
+  }
+}
+```
+
+## Remove Annoying Views
+
+This workaround provides ways to remove annoying views from the player that can't be removed with official APIs.
+
+### Hide Title
+
+Hide title and channel picture at once
+
+Add this to `ayp_youtube_player.html`, and call it inside `onReady`.
+
+```js
+function hideVideoTitle() {
+  setInterval(() => {
+    const playerIFrame = document.querySelector("iframe");
+    if (!playerIFrame) {
+      return;
+    }
+    
+    const frameDoc = playerIFrame.contentDocument;
+    if (!frameDoc) {
+      return;
+    }
+
+    const title = frameDoc.querySelector('.ytp-chrome-top');
+    if (title) {
+      title.style.display = 'none';
+    }
+  }, 100);
+}
+```
+
+### Hide 'More Videos' section that covers most of the video when paused (Only visible on tablets and bigger screens)
+
+Add this to `ayp_youtube_player.html`, and call it inside `onReady`.
+
+```js
+function hideTabletPopup() {
+  setInterval(() => {
+    const playerIFrame = document.querySelector("iframe");
+    if (!playerIFrame) {
+      return;
+    }
+
+    const frameDoc = playerIFrame.contentDocument;
+    if (!frameDoc) {
+      return;
+    }
+
+    const collection = frameDoc.getElementsByClassName("ytp-pause-overlay-container")[0];
+    if (!collection) {
+      return;
+    }
+    collection.style.display = 'none';
+  }, 100);
+}
+```
+## Hide Captions
+
+Add this to `ayp_youtube_player.html`, and call it inside `onReady`.
+
+```js
+function hideCaption() {
+  setInterval(() => {
+    if(!player) {
+      return;
+    }
+    player.unloadModule('captions');
+  }, 1000);
+}
+```
+
+To enable captions
+
+```js
+function hideCaption() {
+  if(!player) {
+    return;
+  }
+  player.loadModule('captions');
+}
+```
+
+## Play Next Video
```

Comment:
it's not super clear to me what this workaround does, can you please try to explain a bit better? why would someone want to use it?

Reply:
that code basically gets YouTube video from "more video"(you can think of it like recommendation  section on YouTube) so if you want to play next recommended video.It is like "auto play" function on YouTube. I personally call that code from `YT.PlayerState.ENDED` so when video ends it automatically plays the next recommended video.

Reply:
Ok that makes sense, thanks for the explanation.